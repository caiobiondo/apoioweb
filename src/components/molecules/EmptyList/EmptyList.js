import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Icon } from 'natura-ui';

import { Container, IconWrapper, Title, Description } from './EmptyList.styles';

const EmptyList = ({ icon, titleId, descriptionId }) => {
  return (
    <Container>
      <IconWrapper>
        <Icon file={icon} />
      </IconWrapper>
      <Title>{titleId && <FormattedMessage id={titleId} />}</Title>
      <Description>{descriptionId && <FormattedMessage id={descriptionId} />}</Description>
    </Container>
  );
};

EmptyList.defaultProps = {
  icon: 'ico_box',
};

EmptyList.propTypes = {
  icon: PropTypes.string.isRequired,
  titleId: PropTypes.string,
  descriptionId: PropTypes.string,
};

export default EmptyList;
