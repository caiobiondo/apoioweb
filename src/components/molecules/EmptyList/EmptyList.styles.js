import styled from 'styled-components';
import { Responsive } from '@entria/components';
import { gray700, gray150 } from 'styles/colors';
import { fs14, fs18, fs20, fs28, fw900, RobotoRegular } from 'styles/typography';
import { spMedium } from 'styles/spacing';
import { Center } from 'styles/mixins';

export const Container = styled.div`
  ${Center};
  padding: 50px 0;
`;

export const IconWrapper = styled.div`
  margin-bottom: 31.5px;
  display: inline-block;

  svg {
    height: 81.5px;
    width: 93px;
    fill: ${gray150};
  }

  @media (max-width: ${Responsive.VIEWPORT.SMALL}px) {
    svg {
      height: 71.5px;
      width: 83px;
    }
  }
`;

export const Title = styled.div`
  color: ${gray700};
  opacity: 0.8;
  font-size: ${fs28};
  font-weight: ${fw900};
  padding-bottom: ${spMedium};
  text-align: center;

  @media (max-width: ${Responsive.VIEWPORT.SMALL}px) {
    font-size: ${fs20};
  }
`;

export const Description = styled.div`
  color: ${gray700};
  opacity: 0.8;
  font-family: ${RobotoRegular};
  font-size: ${fs18};
  text-align: center;

  @media (max-width: ${Responsive.VIEWPORT.SMALL}px) {
    font-size: ${fs14};
  }
`;
