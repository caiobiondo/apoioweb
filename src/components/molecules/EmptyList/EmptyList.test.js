import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import EmptyList from './EmptyList';

describe('EmptyList', () => {
  it('should render a default empty list indicator', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<EmptyList />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render an empty orders indicator', () => {
    const renderer = new ShallowRenderer();

    renderer.render(
      <EmptyList icon="ico_box" titleId="ordersEmptyList" descriptionId="ordersWithoutOrders" />,
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render an empty customers indicator', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<EmptyList icon="ico_box" titleId="customersEmptyList" />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
