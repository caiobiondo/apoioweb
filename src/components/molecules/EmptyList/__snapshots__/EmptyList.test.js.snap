// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList should render a default empty list indicator 1`] = `
<EmptyListstyles__Container>
  <EmptyListstyles__IconWrapper>
    <Icon
      file="ico_box"
    />
  </EmptyListstyles__IconWrapper>
  <EmptyListstyles__Title />
  <EmptyListstyles__Description />
</EmptyListstyles__Container>
`;

exports[`EmptyList should render an empty customers indicator 1`] = `
<EmptyListstyles__Container>
  <EmptyListstyles__IconWrapper>
    <Icon
      file="ico_add_customer"
    />
  </EmptyListstyles__IconWrapper>
  <EmptyListstyles__Title>
    <FormattedMessage
      id="customersEmptyList"
      values={Object {}}
    />
  </EmptyListstyles__Title>
  <EmptyListstyles__Description>
    <FormattedMessage
      id="customersEmptyListDescription"
      values={Object {}}
    />
  </EmptyListstyles__Description>
</EmptyListstyles__Container>
`;

exports[`EmptyList should render an empty orders indicator 1`] = `
<EmptyListstyles__Container>
  <EmptyListstyles__IconWrapper>
    <Icon
      file="ico_box"
    />
  </EmptyListstyles__IconWrapper>
  <EmptyListstyles__Title>
    <FormattedMessage
      id="ordersEmptyList"
      values={Object {}}
    />
  </EmptyListstyles__Title>
  <EmptyListstyles__Description>
    <FormattedMessage
      id="ordersWithoutOrders"
      values={Object {}}
    />
  </EmptyListstyles__Description>
</EmptyListstyles__Container>
`;
